addon/-private/system/model/record-data-map.ts

```diff
@@ -0,0 +1,54 @@
+import { DEBUG } from '@glimmer/env';
+
+export const RECORD_DATA_KEY = `${Date.now()}-record-data`;
```

Comment:
Do we want to merge this map before we get rid of the internal model mapping? Is it easy to measure the perf impact?

Reply:
> There isn’t one

@runspired how do we know this?